### PR TITLE
Add an updateRange method.

### LINKED ISSUE #67: Updating month range causes visual glitches (e.g. the calendar to animate in)
Updating the calendar range calls `notifyDataSetChange` on the adapter making it difficult for the `RecyclerView`  to update only what's necessary (even though stable IDs are on). This leads to the calendar animating in some cases even though the range change doesn't affect the currently displayed month.

This issue makes it impossible to seamlessly extend the calendar during interaction (e.g. to simulate an "endless" calendar by adding more months as the user is scrolling).

The solution is to use more fine-grained updates with `notifyItemRange*` methods. I should be able to submit a change soon that adds a new `updateRange` method with fine-grained updates.

### PR DESCRIPTION
**Pull request by msimonides**
The method changes startMonth and/or endMonth with minimal updates
to the underlying RecyclerView.
This fixes #67.

### CODE REVIEW

File: library/src/main/java/com/kizitonwose/calendarview/CalendarView.kt [hunk 1]
```diff
@@ -5,13 +5,15 @@ import android.util.AttributeSet
 import android.view.View.MeasureSpec.UNSPECIFIED
 import android.view.ViewGroup
 import androidx.annotation.Px
+import androidx.recyclerview.widget.DiffUtil
```
Comment (kbwatts) [inline library/src/main/java/com/kizitonwose/calendarview/CalendarView.kt]:
```suggestion
import androidx.recyclerview.widget.DiffUtiljhghghgf
```

File: library/src/main/java/com/kizitonwose/calendarview/CalendarView.kt [hunk 2]
```diff
@@ -619,6 +622,49 @@ open class CalendarView : RecyclerView {
...
+                outDateStyle, inDateStyle, maxRowCount,
+                startMonth, endMonth, firstDayOfWeek ?: return,
+                hasBoundaries
+            )
+        DiffUtil.calculateDiff(
+            MonthRangeDiffCallback(oldStartMonth, oldEndMonth, startMonth, endMonth), false)
+            .dispatchUpdatesTo(calendarAdapter)
+    }
+
+    private class MonthRangeDiffCallback(
```
Comment (kizitonwose) [inline library/src/main/java/com/kizitonwose/calendarview/CalendarView.kt]: This diffing logic will not work if the calendar's `maxRowCount` value is not set to 6. For example, in a week mode calendar(`maxRowCount` is 1), a `YearMonth` will be represented over about 4 or 5 indices so getting the list count by counting the number of months between the start and end months will give an inaccurate result.

A better approach would be to use the old and the newly created `MonthConfig` for diffing, There's a `months` property in it which gives the accurate size.

Comment (msimonides) [reply library/src/main/java/com/kizitonwose/calendarview/CalendarView.kt]: Good point, done.

**Comment by kizitonwose** [comment]
Thanks for contributing!

**Pull request merged**